transaction_test.go

```diff
@@ -1672,3 +1678,73 @@ func TestInfiniteTransaction(t *testing.T) {
 
 	require.True(t, fvmerrors.IsComputationLimitExceededError(result.Error))
 }
+
+func TestSubmitTransactionWithCustomLogger(t *testing.T) {
+
+	t.Parallel()
+	var memlog bytes.Buffer
+	memlogWrite := io.Writer(&memlog)
+	logger := zerolog.New(memlogWrite).Level(zerolog.DebugLevel)
+
+	b, err := emulator.NewBlockchain(
+		emulator.WithStorageLimitEnabled(false),
+		emulator.WithLogger(logger),
+	)
+	require.NoError(t, err)
+
+	addTwoScript, _ := deployAndGenerateAddTwoScript(t, b)
+
+	tx1 := flow.NewTransaction().
+		SetScript([]byte(addTwoScript)).
+		SetGasLimit(flowgo.DefaultMaxTransactionGasLimit).
+		SetProposalKey(b.ServiceKey().Address, b.ServiceKey().Index, b.ServiceKey().SequenceNumber).
+		SetPayer(b.ServiceKey().Address).
+		AddAuthorizer(b.ServiceKey().Address)
+
+	signer, err := b.ServiceKey().Signer()
+	require.NoError(t, err)
+
+	err = tx1.SignEnvelope(b.ServiceKey().Address, b.ServiceKey().Index, signer)
+	require.NoError(t, err)
+
+	// Submit tx1
+	err = b.AddTransaction(*tx1)
+	assert.NoError(t, err)
+
+	// Execute tx1
+	result, err := b.ExecuteNextTransaction()
+	assert.NoError(t, err)
+	assertTransactionSucceeded(t, result)
+
+	_, err = b.CommitBlock()
+	assert.NoError(t, err)
+
+	// tx1 status becomes TransactionStatusSealed
+	tx1Result, err := b.GetTransactionResult(tx1.ID())
+	assert.NoError(t, err)
+	assert.Equal(t, flow.TransactionStatusSealed, tx1Result.Status)
+
+	var meter Meter
+	scanner := bufio.NewScanner(&memlog)
+	for scanner.Scan() {
+		txt := scanner.Text()
+		if strings.Contains(txt, "transaction execution data") {
+			err = json.Unmarshal([]byte(txt), &meter)
+		}
+	}
+
+	assert.NoError(t, err)
+	assert.Equal(t, `{LedgerInteractionUsed:9373 ComputationUsed:18 MemoryUsed:0 ComputationIntensities:map[Statement:10 Loop:1 FunctionInvocation:8 CreateCompositeValue:2 TransferCompositeValue:3 CreateArrayValue:4 TransferArrayValue:446 CreateDictionaryValue:1 ComputationKind(2005):1 ComputationKind(2006):1 ComputationKind(2007):1 ComputationKind(2008):1 ComputationKind(2011):2 ComputationKind(2015):2 ComputationKind(2017):2 ComputationKind(2020):4760 ComputationKind(2022):1 ComputationKind(2025):3 ComputationKind(2026):1621 ComputationKind(2027):1] MemoryIntensities:map[]}`, fmt.Sprintf("%+v", meter))
```

Comment:
This is a bit strict I think. I had something like this in mind:
```suggestion
assert.Greater(t, meter. LedgerInteractionUsed, 0)
assert.Greater(t, meter. ComputationUsed, 0)
assert.Equal(t, meter. MemoryUsed, 0) // this will change with the new emulator version I think
assert.Greater(t, meter. ComputationIntensities.len, 0)
assert.Equal(t, meter. MemoryIntensities.len, 0) // this will change with the new emulator version I think
```